commit 7354012d871a98cfeba6ab962af30b16d0455e5f (refs/changes/19/2919/2)
Author: Jerry Jelinek <jerry.jelinek@joyent.com>
Date:   2017-11-07T21:42:26+00:00 (1 year, 11 months ago)
    
    OS-6435 cred reference count leak leads to zone livelock
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
    Approved by: Patrick Mooney <patrick.mooney@joyent.com>

diff --git a/usr/src/uts/common/inet/squeue.c b/usr/src/uts/common/inet/squeue.c
index 33bc43ca9a..a1c0dbe697 100644
--- a/usr/src/uts/common/inet/squeue.c
+++ b/usr/src/uts/common/inet/squeue.c
@@ -1500,6 +1500,9 @@ squeue_try_drain_one(squeue_t *sqp, conn_t *compare_conn)
 	CONN_DEC_REF(connp);
 	SQUEUE_DBG_CLEAR(sqp);
 
+	if (ira != NULL)
+		ira_cleanup(ira, B_TRUE);
+
 done:
 	mutex_enter(&sqp->sq_lock);
 	sqp->sq_state &= ~(SQS_PROC);
